dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsFetcherTest.java

```diff
@@ -0,0 +1,157 @@
+    }
+    
+    @Test
+    public void loadSettings__NOT_EXISTING_PROPERTIES_FROM_CLASSPATH() {
+        
+        // given
+        FileSettingLocations settingLocations = new FileSettingLocations();
+        settingLocations.addLocation(SettingLocationType.CLASSPATH, "/notexisting.properties", false);
+        
+        FileBasedSettingsFetcher settingsFetcher = new FileBasedSettingsFetcher(settingLocations);
+        
+        // when
+        Executable loadSettingsOperation = () -> settingsFetcher.loadSettings();
+        
+        // then
+        assertThrows(RuntimeException.class, loadSettingsOperation);
+    }
+    
+    @Test
+    public void loadSettings__NOT_EXISTING_OPTIONAL_PROPERTIES_FROM_CLASSPATH() {
+        
+        // given
+        FileSettingLocations settingLocations = new FileSettingLocations();
+        settingLocations.addLocation(SettingLocationType.CLASSPATH, "/notexisting.properties", true);
+        
+        FileBasedSettingsFetcher settingsFetcher = new FileBasedSettingsFetcher(settingLocations);
+        
+        // when
+        settingsFetcher.loadSettings();
+        
+        // then
+        assertEquals(0, settingsFetcher.getAllSettings().size());
+    }
+    
+    
+    @Test
+    public void loadSettings__PROPERTIES_FROM_FILESYSTEM(@TempDir Path tempDir) throws IOException {
+        
+        // given
+        byte[] propertiesBytes = IOUtils.resourceToByteArray("/test1.properties");
+        File filesystemPropertiesFile = new File(tempDir.toFile(), "filesystem.properties");
+        FileUtils.writeByteArrayToFile(filesystemPropertiesFile, propertiesBytes);
+        
+        FileSettingLocations settingLocations = new FileSettingLocations();
+        settingLocations.addLocation(SettingLocationType.FILESYSTEM, filesystemPropertiesFile.getAbsolutePath(), false);
+        
+        FileBasedSettingsFetcher settingsFetcher = new FileBasedSettingsFetcher(settingLocations);
+        
+        // when
+        settingsFetcher.loadSettings();
+        
+        // then
+        assertEquals("keyval1", settingsFetcher.getSetting("BuiltinUsers.KEY"));
+        assertEquals("someval1", settingsFetcher.getSetting(":SomeSetting"));
+        assertEquals("testval1", settingsFetcher.getSetting(":Test1Setting"));
+        assertThat(settingsFetcher.getAllSettings().keySet(),
+                containsInAnyOrder("BuiltinUsers.KEY", ":SomeSetting", ":Test1Setting"));
+    }
+    
+    @Test
+    public void loadSettings__NOT_EXISTING_PROPERTIES_FROM_FILESYSTEM() {
+        
+        // given
+        FileSettingLocations settingLocations = new FileSettingLocations();
+        settingLocations.addLocation(SettingLocationType.FILESYSTEM, "notexisting.properties", false);
+        
+        FileBasedSettingsFetcher settingsFetcher = new FileBasedSettingsFetcher(settingLocations);
+        
+        // when
+        Executable loadSettingsOperation = () -> settingsFetcher.loadSettings();
```

Comment:
Isn't `settingsFetcher::loadSettings` more readable?

Reply:
Ok. I will change it